tests/MongoDB.Driver.Core.Tests/Specifications/connection-string/TestRunner.cs

```diff
@@ -60,6 +58,172 @@ public void RunTestDefinition(JsonDrivenTestCase testCase)
             }
         }
 
+        private void AssertBoolean(bool? @bool, BsonValue expectedValue)
+        {
+            @bool.Should().Be(expectedValue.IsBsonNull ? (bool?)null : expectedValue.ToBoolean());
+        }
+
+        private void AssertDictionary(IReadOnlyDictionary<string, string> dict, BsonDocument expectedDocument)
+        {
+            var sortedDictionary = dict
+                .OrderBy(pair => pair.Key)
+                .ToDictionary(pair => pair.Key, pair => pair.Value);
+            var bsonDocument = new BsonDocument();
+            foreach (var item in sortedDictionary)
+            {
+                bsonDocument.Add(item.Key, ConvertToBsonValue(item.Value));
+            }
+
+            var expectedDocumentElements = expectedDocument
+                .Elements
+                .OrderBy(e => e.Name);
+            var sortedExpectedDocument = new BsonDocument(expectedDocumentElements);
+
+            sortedExpectedDocument.Should().Be(bsonDocument);
+        }
+
+        private void AssertEnum<T>(T? @enum, BsonValue expectedValue) where T : struct
+        {
+            @enum.Should().Be(expectedValue.IsBsonNull ? null : (T?)Enum.Parse(typeof(T), expectedValue.AsString, true));
+        }
+
+        private void AssertOptions(ConnectionString connectionString, BsonDocument definition)
+        {
+            if (definition.TryGetValue("options", out var options) && !options.IsBsonNull)
+            {
+                foreach (var expectedOption in options.AsBsonDocument.Elements)
+                {
+                    var lowerName = expectedOption.Name.ToLower();
+                    switch (lowerName)
+                    {
+                        case "appname":
+                            connectionString.ApplicationName.Should().Be(expectedOption.Value.AsString);
+                            break;
+                        case "authmechanism":
+                            connectionString.AuthMechanism.Should().Be(expectedOption.Value.AsString);
+                            break;
+                        case "authmechanismproperties":
+                            AssertDictionary(connectionString.AuthMechanismProperties, expectedOption.Value.AsBsonDocument);
+                            break;
```

Comment:
Refactor to:

    case "authmechanismproperties":
        var authMechanismProperties = new BsonDocument(connectionString.AuthMechanismProperties.Select(kv => new BsonElement(kv.Key, ConvertToBsonValue(kv.Value))));
        authMechanismProperties.Should().BeEquivalentTo(expectedOption.Value.AsBsonDocument);
        break;

Reply:
Note: `Should().BeEquivalentTo(...)` ignores order as required by this assertion.